virttest/utils_misc.py

```diff
@@ -3025,7 +3025,7 @@ def get_ksmtuned_pid(self):
         try:
             utils_path.find_command("ksmtuned")
         except utils_path.CmdNotFoundError:
-            raise KSMTunedNotSupportedError
```

Comment:
I suggest that add the step to cancel it in the upper caller, like:
test_setup.py
```
        # Check if ksmtuned is running before the test
        try:
            self.ksmtuned_process = self.ksmctler.get_ksmtuned_pid()
        except utils_misc.KSMTunedNotSupportedError:
            raise exceptions.TestCancel("ksmtuned command not found.")
```
I think this API should not be related with test case level, it provides such method for caller, but the behaviours about how to handle the error should be determined by the upper caller, and we can not make sure that all the test case scenarios can cancel test due to command not found.
For the previous design(`raise KSMTunedNotSupportedError`), we can keep the method independence which doesn't coupling with other upper level.

Reply:
Hi, @yanan-fu 
Would you like to help to review it? Thanks.

Reply:
+1, it is test_setup's response as my understanding.

Reply:
Although, one more high level checking is in env_process:
https://github.com/avocado-framework/avocado-vt/blob/62ae82c9848dd8a102e7b3de9b166c03268d237b/virttest/env_process.py#L989

cancel test with the test env is not suitable for our testing,  i think env_process is better than test_setup and then utils_misc.

Reply:
@YongxueHong  What's your opinion about this ?

Reply:
@yanan-fu 

So only adding in [ksm_base.cfg](https://github.com/autotest/tp-qemu/blob/master/qemu/tests/cfg/ksm_base.cfg) the entry: `cmds_installed_host = "ksmtuned"` will be valid?

I've tested it and if ksmtuned is not installed reports the following:

 (1/3) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.ksm_base.disable.other_host.q35: ERROR: 'CmdNotFoundError' object has no attribute 'message' (0.20 s)

Reply:
The reason is python3 `Exception` does not support `message` function.  This is another issue not related with this patch directly,  fix in env_process, you can try:
```
    if params.get("cmds_installed_host"):
        for cmd in params.get("cmds_installed_host").split():
            try:
                path.find_command(cmd)
            except path.CmdNotFoundError as msg:
                raise exceptions.TestSkipError(msg)
```

Reply:
Changed! Will update test results after applying both patches